climada/hazard/tc_tracks_synth.py

```diff
@@ -99,15 +143,23 @@ def calc_perturbed_trajectories(tracks,
         Random number generator seed for replicability of random walk.
         Put negative value if you don't want to use it. Default: configuration file.
     decay : bool, optional
-        Whether to compute land decay in probabilistic tracks. Default: True.
+        Whether to apply landfall decay in probabilistic tracks. Default: True.
+    use_global_decay_params : bool, optional
+        Whether to use precomputed global parameter values for landfall decay
+        obtained from IBTrACS (1980-2019). If False, parameters are fitted
+        using historical tracks in input parameter 'tracks', in which case the
+        landfall decay applied depends on the tracks passed as an input and may
+        not be robust if few historical tracks make landfall in this object.
+        Default: True.
     """
+    
     LOGGER.info('Computing %s synthetic tracks.', nb_synth_tracks * tracks.size)
 
     if seed >= 0:
         np.random.seed(seed)
 
     # ensure tracks have constant time steps
-    time_step_h = np.unique([np.unique(x['time_step']) for x in tracks.data])
+    time_step_h = np.unique([np.unique(x['time_step']) for x in tracks.data if x.time.size > 1])
     if not np.allclose(time_step_h, time_step_h[0]):
         LOGGER.error('Tracks have different temporal resolution. '
                      'Please ensure constant time steps by applying equal_timestep beforehand')
```

Comment:
ok, now I am bit confused about this line actually. Why would you use `np.allclose()`? What you expect is that `time_step_h` is a unique defined value (a float). As soon as it is a list, it means that there are two different timestep values. 

The following suggestion would replace line 162 - 167 (I was not able to include lines 166,167 in the the suggestion, so if you like it please remove these manually):
```suggestion
    try:
        [time_step_h] = np.unique([np.unique(x['time_step']) for x in tracks.data])
    except ValueError:
         raise ValueError('Not all tracks have the same temporal resolution. '
                     'Please ensure constant time steps by applying equal_timestep beforehand')
```

Reply:
That would likely work as well, yes. I'm not sure I understand why `np.allclose()` is not appropriate though (to me it sounds like exactly what's needed, namely checking that all values are very close to each other)? I'm fine with your suggestion, but would you mind just enlightening me on that one, so I can recall for next time? :-)

Reply:
I agree that there is not much of a difference, but for me the suggestion is clearer in what this bit of code is doing. In the previous version (np.allclose), the purpose of the blob of code is harder to understand at one glance.